src/wazuh_modules/syscollector/syscollector_bsd.c

```diff
@@ -241,13 +244,13 @@ char* sys_parse_pkg(const char * app_folder, const char * timestamp, int random_
         cJSON_AddItemToObject(object, "program", package);
         cJSON_AddStringToObject(package, "format", "pkg");
 
-        // Check if valid Info.plist file (not an Apple binary property list)
+        // Check if valid Info.plist file
         if (fgets(read_buff, OS_MAXSTR - 1, fp) != NULL) {
-            if (strncmp(read_buff, "<?xml", 5)) {   // Invalid file
-                mtdebug1(WM_SYS_LOGTAG, "Unable to read package information from '%s' (invalid format)", filepath);
-                invalid = 1;
-            } else {
-                // Valid Info.plist file
+            if (!memcmp(read_buff, "bplist00", 8)) {  // Apple binary plist
+                sys_convert_bin_plist(&fp, (char *)read_buff, filepath);
```

Comment:
Should an error arise within `sys_convert_bin_plist`, `read_buf` won't be overwritten (will still contain "_bplist00_"), and the flow will end up in the else statement: https://github.com/wazuh/wazuh/blob/90856778988e7fc905508f5c13f5bce711396b23/src/wazuh_modules/syscollector/syscollector_bsd.c#L393-L396
I though it would be simpler to just re-use the generic error mentioned above.